src/btree/bt_sync.c

```diff
@@ -26,20 +26,23 @@ __sync_checkpoint_can_skip(WT_SESSION_IMPL *session, WT_REF *ref)
     txn = session->txn;
 
     /*
-     * We can skip some dirty pages during a checkpoint. The requirements:
+     * We may skip dirty pages during a checkpoint unless one of the following conditions is met:
```

Comment:
I have reworded this entire section, can I have a thorough review here, please?
Furthermore, why are we not skipping internal pages? The explanation is missing.
@kommiharibabu @bvpvamsikrishna